### PR TITLE
flashback: flashback cluster support tso

### PR DESCRIPTION
**Pull request by benmaoer**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-flashback-cluster.md [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
 
-TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到过去指定的时间点；如果你想要指定一个精确的时间点，可在 `FLASHBACK CLUSTER TO TIMESTAMP` 中使用日期时间和时间函数，日期时间的格式为："2016-10-08 16:45:26.999"，最小时间精度范围为毫秒，通常可只写到秒，例如 "2016-10-08 16:45:26"。
+
+TiDB v6.5.6 开始引入了 `FLASHBACK CLUSTER TO TSO` 的语法，支持更加精确的指定恢复时间戳 [TSO](https://docs.pingcap.com/tidb/stable/tso)，实现更加灵活的数据恢复。
 
 > **注意：**
 >
-> `FLASHBACK CLUSTER TO TIMESTAMP` 是用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
+> `FLASHBACK CLUSTER TO { TIMESTAMP | TSO }` 是用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply sql-statements/sql-statement-flashback-cluster.md:15]: 参考
image

File: sql-statements/sql-statement-flashback-cluster.md [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
 FLASHBACK CLUSTER TO TIMESTAMP '2022-09-21 16:02:50';
+FLASHBACK CLUSTER TO TSO '445494839813079041';
 ```
 
 ### 语法图
 
 ```ebnf+diagram
 FlashbackToTimestampStmt ::=
-    "FLASHBACK" "CLUSTER" "TO" "TIMESTAMP" stringLit
+    'FLASHBACK' 'CLUSTER' 'TO' '{ TIMESTAMP | TSO }' stringLit
```
Comment (qiancai) [inline sql-statements/sql-statement-flashback-cluster.md:28]:
```suggestion
	"FLASHBACK" "CLUSTER" "TO" "TIMESTAMP" stringLit
|	"FLASHBACK" "CLUSTER" "TO" "TSO" LengthNum
```

File: sql-statements/sql-statement-flashback-cluster.md [hunk 3]
```diff
@@ -74,6 +78,52 @@ mysql> SELECT * FROM t;
...
++------+
+|    1 |
++------+
+1 row in set (0.01 sec)
+
+
+mysql> begin;
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> select @@tidb_current_ts;
```
Comment (qiancai) [inline sql-statements/sql-statement-flashback-cluster.md:99]:
```suggestion
mysql> select @@tidb_current_ts;  # 获取当前 TSO
```

File: sql-statements/sql-statement-flashback-cluster.md [hunk 4]
```diff
@@ -34,12 +38,12 @@ FlashbackToTimestampStmt ::=
...
 * 在 `FLASHBACK` 指定的时间点到开始执行的时间段内不能存在相关表结构变更的 DDL 记录。若存在，TiDB 会拒绝该 DDL 操作。
-* 在执行 `FLASHBACK CLUSTER TO TIMESTAMP` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
-* `FLASHBACK CLUSTER TO TIMESTAMP` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
+* 在执行 `FLASHBACK CLUSTER` 前，TiDB 会主动断开所有相关表上的连接，并禁止对这些表进行读写操作，直到 `FLASHBACK` 完成。
+* `FLASHBACK CLUSTER` 命令不能取消，一旦开始执行 TiDB 会一直重试，直到成功。
 
 ## 示例
 
-恢复新插入的数据：
+闪回到指定 Timestamp 恢复新写入的数据：
```
Comment (qiancai) [inline sql-statements/sql-statement-flashback-cluster.md:46]:
```suggestion
闪回到指定的 TIMESTAMP 恢复新写入的数据：
```

File: sql-statements/sql-statement-flashback-cluster.md [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
 > **注意：**
 >
-> `FLASHBACK CLUSTER TO TIMESTAMP` 是用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
+> `FLASHBACK CLUSTER TO { TIMESTAMP | TSO }` 是用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
 
 ## 语法
 
 ```sql
 FLASHBACK CLUSTER TO TIMESTAMP '2022-09-21 16:02:50';
+FLASHBACK CLUSTER TO TSO '445494839813079041';
```
Comment (BornChanger) [inline sql-statements/sql-statement-flashback-cluster.md:21]:
```suggestion
FLASHBACK CLUSTER TO TSO 445494839813079041;
```

File: sql-statements/sql-statement-flashback-cluster.md [hunk 6]
```diff
@@ -74,6 +78,52 @@ mysql> SELECT * FROM t;
...
+
+mysql> rollback;
+Query OK, 0 rows affected (0.00 sec)
+
+
+mysql> DELETE FROM t;
+Query OK, 1 rows affected (0.00 sec)
+
+
+mysql> FLASHBACK CLUSTER TO TSO '446113975683252225';
```
Comment (BornChanger) [inline sql-statements/sql-statement-flashback-cluster.md:115]:
```suggestion
mysql> FLASHBACK CLUSTER TO TSO 446113975683252225;
```

File: sql-statements/sql-statement-flashback-cluster.md (hunk 2 above)
Comment (qiancai) [reply sql-statements/sql-statement-flashback-cluster.md:28]: 参考 https://github.com/pingcap/tidb/blob/0c7659c1907dd033af8b13b15eee1960a0d9ee79/pkg/parser/parser.y#L3005

File: sql-statements/sql-statement-flashback-cluster.md [hunk 7]
```diff
@@ -1,27 +1,32 @@
...
+FLASHBACK CLUSTER TO TSO '445494839813079041';
 ```
 
 ### 语法图
 
 ```ebnf+diagram
 FlashbackToTimestampStmt ::=
-    "FLASHBACK" "CLUSTER" "TO" "TIMESTAMP" stringLit
+	"FLASHBACK" "CLUSTER" "TO" "TIMESTAMP" stringLit
+|	"FLASHBACK" "CLUSTER" "TO" "TSO" LengthNum
```
Comment (qiancai) [inline sql-statements/sql-statement-flashback-cluster.md:29]:
```suggestion
	‘FLASHBACK‘ ‘CLUSTER‘ ‘TO‘ ‘TIMESTAMP‘ stringLit
|	’FLASHBACK’ ’CLUSTER’ ’TO’ ’TSO’ LengthNum
```

**Review by BornChanger — APPROVED** [review]
Rest LGTM.

File: sql-statements/sql-statement-flashback-cluster.md [hunk 8]
```diff
@@ -74,6 +79,52 @@ mysql> SELECT * FROM t;
...
++------+
+|    1 |
++------+
+1 row in set (0.01 sec)
+
+
+mysql> begin;
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> select @@tidb_current_ts;  --  获取当前 TSO
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-cluster.md:100]:
```suggestion
mysql> SELECT @@tidb_current_ts;  --  获取当前 TSO
```

File: sql-statements/sql-statement-flashback-cluster.md [hunk 9]
```diff
@@ -74,6 +79,52 @@ mysql> SELECT * FROM t;
...
++--------------------+
+| @@tidb_current_ts  |
++--------------------+
+| 446113975683252225 |
++--------------------+
+1 row in set (0.00 sec)
+
+mysql> rollback;
```
Comment (hfxsd) [inline sql-statements/sql-statement-flashback-cluster.md:108]:
```suggestion
mysql> ROLLBACK;
```

**Review by hfxsd — APPROVED** [review]

File: sql-statements/sql-statement-flashback-cluster.md [hunk 10]
```diff
@@ -1,27 +1,32 @@
...
+title: FLASHBACK CLUSTER
+summary: TiDB 数据库中 FLASHBACK CLUSTER 的使用概况。
+aliases: ['/zh/tidb/v6.5/sql-statement-flashback-to-timestamp']
 ---
 
-# FLASHBACK CLUSTER TO TIMESTAMP
+# FLASHBACK CLUSTER
 
-TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到特定的时间点。
+TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到过去指定的时间点。指定时间点时，你可以使用日期时间和时间函数，日期时间的格式为："2016-10-08 16:45:26.999"，最小时间精度范围为毫秒，通常可只写到秒，例如 "2016-10-08 16:45:26"。
```
Comment (qiancai) [inline sql-statements/sql-statement-flashback-cluster.md:9]:
```suggestion
TiDB v6.4.0 引入了 `FLASHBACK CLUSTER TO TIMESTAMP` 语法，其功能是将集群的数据恢复到过去指定的时间点。指定时间点时，你可以使用日期时间和时间函数，日期时间的格式为：'2016-10-08 16:45:26.999'，最小时间精度范围为毫秒，通常可只写到秒，例如 "2016-10-08 16:45:26"。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**